subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftCachingIntegrationTest.groovy

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibraries
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Unroll
+
+@Requires(TestPrecondition.SWIFT_SUPPORT)
+class SwiftCachingIntegrationTest extends AbstractInstalledToolChainIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    def app = new SwiftAppWithLibraries()
+
+    def setupProject(TestFile project = temporaryFolder.testDirectory) {
+        project.file('settings.gradle') << '''
+            include 'hello', 'log'
+            rootProject.name = 'app'
+        '''
+        project.file('settings.gradle') << localCacheConfiguration()
+        project.file('build.gradle').text = '''
+            apply plugin: 'swift-executable'
+            dependencies {
+                implementation project(':hello')
+            }
+
+            project(':hello') {
+                apply plugin: 'swift-library'
+                dependencies {
+                    api project(':log')
+                }
+            }
+            project(':log') {
+                apply plugin: 'swift-library'
+            }
+        '''
+        app.logLibrary.writeToProject(project.file('log'))
+        app.library.writeToProject(project.file('hello'))
+        app.executable.writeToProject(project)
+    }
+
+    def 'compilation can be cached'() {
+        setupProject()
+
+        when:
+        withBuildCache().run compileTask(buildType)
+
+        then:
+        compileIsNotCached(buildType)
```

Comment:
Let's check the dependency tasks, too. It would be nice to have something like `compileIsNotCached(":hello", buildType)` etc.

Reply:
fixed.

Reply:
I think testing if the dependencies are cached/not-cached is not essential for this test.

Reply:
I think this is a bit confusing. It would be somewhat better if we did tested the dependency tasks as well. It would be much better if we had a single-project test case for this one.

Reply:
I am now asserting the outcome of all compile tasks.